lib/sepa_king/validator.rb

```diff
@@ -31,7 +31,7 @@ def validate(record)
   end
 
   class CreditorIdentifierValidator < ActiveModel::Validator
-    REGEX = /\A[a-zA-Z]{2,2}[0-9]{2,2}([A-Za-z0-9]|[\+|\?|\/|\-|\:|\(|\)|\.|,|']){3,3}([A-Za-z0-9]|[\+|\?|\/|\-|:|\(|\)|\.|,|']){1,28}\z/
+    REGEX = %r{\A[a-zA-Z]{2,2}[0-9]{2,2}[A-Za-z0-9+?/-:().,']{3,3}[A-Za-z0-9+?/-:().,']{1,28}\z}
```

Comment:
You have to either escape `-` or move it to the very end of the character class, i.e.:
```suggestion
    REGEX = %r{\A[a-zA-Z]{2,2}[0-9]{2,2}[A-Za-z0-9+?/:().,'-]{3,3}[A-Za-z0-9+?/:().,'-]{1,28}\z}
```
Otherwise it will be treated as a character range just like `a-z`, i.e. `/-:` would attempt to match all characters from `/` to `:` which may lead to unexpected results.

Another thing I just noticed is that the latter two character classes are in fact identical:
```ruby
REGEX = %r{\A
  [a-zA-Z]{2,2}
  [0-9]{2,2}
  [A-Za-z0-9+?/:().,'-]{3,3}   # < have the same
  [A-Za-z0-9+?/:().,'-]{1,28}  # < characters
\z}x
```
However, If I understand the specs correctly, this isn't quite right. The 3-character part after the two check digits is the "creditor business code" which is _alphanumeric_.

The regex should probably be: (with simplified repetition)

```ruby
REGEX = %r{\A
  [a-zA-Z]{2}                 # ISO country code
  [0-9]{2}                    # Check digits
  [A-Za-z0-9]{3}              # Creditor business code
  [A-Za-z0-9+?/:().,'-]{1,28} # National identifier
\z}x
```

Reply:
You are totally correct, thanks again. I have fixed this.